internal/server/data/destination.go

```diff
@@ -221,3 +222,103 @@ func CountDestinationsByConnectedVersion(tx ReadTxn) ([]DestinationsCount, error
 func CountAllDestinations(tx ReadTxn) (int64, error) {
 	return countRows(tx, destinationsTable{})
 }
+
+type DestinationAccess struct {
+	UserID           uid.ID
+	UserSSHLoginName string
+	Privilege        string
+	Resource         string
+}
+
+type ListDestinationAccessResult struct {
+	// Items is the list of items that match the query, excluding soft-deleted rows.
+	Items []DestinationAccess
+
+	// MaxUpdateIndex is the maximum update_index from all the grants and groups
+	// that match the query. This MUST include soft-deleted rows as well.
```

Comment:
Why does it have to include soft-deleted rows, is it because that ensures clients get updated when a row gets deleted?